File: app/src/main/java/me/liuyun/bjutlgn/WiFiApplication.java
package me.liuyun.bjutlgn;

import android.app.Application;
import android.content.SharedPreferences;
import android.content.res.Resources;
import android.preference.PreferenceManager;

import me.liuyun.bjutlgn.db.FlowManager;
import me.liuyun.bjutlgn.util.ThemeHelper;

public class WiFiApplication extends Application {
    private FlowManager flowManager;
    private Resources res;
    private SharedPreferences prefs;

    @Override
    public void onCreate() {
        super.onCreate();
        prefs = PreferenceManager.getDefaultSharedPreferences(this);
        res = getResources();
        flowManager = new FlowManager(this);
        ThemeHelper.getInstance().init(this, prefs.getInt("theme", R.style.ThemeBlue));
    }

    public FlowManager getFlowManager() {
        return flowManager;
    }

    public Resources getRes() {
        return res;
    }

    public SharedPreferences getPrefs() {
        return prefs;
    }
}

File: app/src/main/kotlin/me/liuyun/bjutlgn/WiFiApplication.kt
package me.liuyun.bjutlgn

import android.app.Application
import android.content.SharedPreferences
import android.content.res.Resources
import android.preference.PreferenceManager
import me.liuyun.bjutlgn.db.FlowManager
import me.liuyun.bjutlgn.util.ThemeHelper

class WiFiApplication : Application() {
    lateinit var flowManager: FlowManager
    lateinit var res: Resources
    lateinit var prefs: SharedPreferences

    override fun onCreate() {
        super.onCreate()
        prefs = PreferenceManager.getDefaultSharedPreferences(this)
        res = resources
        flowManager = FlowManager(this)
        ThemeHelper.init(this, prefs.getInt("theme", R.style.ThemeBlue))
    }
}
